analyses/snv-callers/lancet-wxs-tests/explore-tcga-pbta.Rmd

```diff
@@ -0,0 +1,591 @@
+  dplyr::filter(!is.na(chr)) %>%
+  dplyr::mutate(chr = paste0("chr", chr)) %>%
+  readr::write_tsv(file.path(ref_dir, "gencode.v19.basic.exome.tsv"),
+    col_names = FALSE
+  )
+```
+
+After using [UCSC BED liftover](https://genome.ucsc.edu/cgi-bin/hgLiftOver) to convert 
+from the Target BED regions for TCGA data from hg19 to hg38 (What the PBTA data is using).
+
+```{r}
+tcga_lift_bed <- readr::read_tsv(file.path(
+  ref_dir,
+  "hg38_liftover_genome_gencode.v19.basic.exome.bed"
+),
+col_names = c("chr", "start", "end")
+) %>%
+  dplyr::mutate(chr = stringr::word(chr, sep = "_", 1))
+
+# Make GRanges for CNV data
+tcga_lift_granges <- GenomicRanges::GRanges(
+  seqnames = tcga_lift_bed$chr,
+  ranges = IRanges::IRanges(
+    start = tcga_lift_bed$start,
+    end = tcga_lift_bed$end
+  )
+)
+
+# Reduce this to it's essential ranges
+tcga_lift_granges <- GenomicRanges::reduce(tcga_lift_granges)
+```
+
+Make a TCGA BED GenomicRanges. 
+
+```{r}
+# Make GRanges for CNV data
+tcga_granges <- GenomicRanges::GRanges(
+  seqnames = tcga_bed$chr,
+  ranges = IRanges::IRanges(
+    start = tcga_bed$start,
+    end = tcga_bed$end
+  )
+)
+
+# Reduce this to it's essential ranges
+tcga_granges <- GenomicRanges::reduce(tcga_granges)
+```
+
+Format the PBTA WXS data as a GRanges object. 
+
+```{r}
+# pbta_bed <- readr::read_tsv(file.path(scratch_dir, "intersect_cds_WXS.bed"),
+pbta_bed <- readr::read_tsv(file.path(data_dir, "WXS.hg38.100bp_padded.bed"),
+  col_names = c("chr", "start", "end")
+) %>%
+  dplyr::filter(!is.na(chr))
+
+# Make GRanges for CNV data
+pbta_granges <- GenomicRanges::GRanges(
+  seqnames = pbta_bed$chr,
+  ranges = IRanges::IRanges(
+    start = pbta_bed$start,
+    end = pbta_bed$end
+  )
+)
+
+# Reduce this to it's essential ranges
+pbta_granges <- GenomicRanges::reduce(pbta_granges)
+```
+
+Read in the coding sequence regions file. 
+
+```{r}
+cds_bed <- readr::read_tsv(file.path(
+  scratch_dir,
+  "gencode.v27.primary_assembly.annotation.bed"
+),
+col_names = c("chr", "start", "end")
+) %>%
+  dplyr::filter(!is.na(chr))
+
+# Make GRanges
+cds_granges <- GenomicRanges::GRanges(
+  seqnames = cds_bed$chr,
+  ranges = IRanges::IRanges(
+    start = cds_bed$start,
+    end = cds_bed$end
+  )
+)
+
+# Reduce this to it's essential ranges
+cds_granges <- GenomicRanges::reduce(cds_granges)
+```
+
+We also need to see how this translates to CDS regions:
+
+```{r}
+# Find CDS intersection
+pbta_cds_granges <- GenomicRanges::intersect(pbta_granges, cds_granges)
+tcga_cds_granges <- GenomicRanges::intersect(tcga_granges, cds_granges)
+tcga_lift_cds_granges <- GenomicRanges::intersect(tcga_lift_granges, cds_granges)
+```
+
+## Make Venn diagrams of these overlaps
+
+Find overlap between the TCGA (non-liftover) and PBTA data. 
+
+```{r}
+bed_overlap(pbta_granges,
+  tcga_granges,
+  plot_name = "PBTA vs TCGA WXS target BED",
+  name_1 = "PBTA",
+  name_2 = "TCGA"
+)
+```
```

Comment:
I don't think this is necessary. We don't expect unlifted data to work, so I would simply eliminate these comparisons completely.